core/src/banking_stage.rs

```diff
@@ -503,29 +503,28 @@ impl BankingStage {
         socket: &UdpSocket,
         hold: bool,
     ) {
-        if enable_forwarding {
-            let next_leader = poh_recorder
-                .lock()
-                .unwrap()
-                .leader_after_n_slots(FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET);
-            next_leader.map_or((), |leader_pubkey| {
-                let leader_addr = {
-                    cluster_info.lookup_contact_info(&leader_pubkey, |leader| leader.tpu_forwards)
-                };
-
-                leader_addr.map_or((), |leader_addr| {
-                    let _ =
-                        Self::forward_buffered_packets(&socket, &leader_addr, &buffered_packets);
-                    if hold {
-                        buffered_packets.retain(|b| b.1.is_empty());
-                        for b in buffered_packets.iter_mut() {
-                            b.2 = true;
-                        }
-                    } else {
-                        buffered_packets.clear();
-                    }
-                })
-            })
+        if !enable_forwarding {
+            buffered_packets.clear(); // TODO: What if hold is true?
+            return;
```

Comment:
These will be votes that came through gossip. Maybe we should not clear here if hold==true. I think `cluster_info_vote_listener` would not re-send them. The votes might be somewhat old by the time the leader slot comes though and replaced by newer ones. Might have to enable some metrics on the real network to see if it's worth it to keep them.

Reply:
updated